### PR TITLE
replicate architecture changes

### LINKED ISSUE #1396: document the new architecture
### Related code PR

_No response_

### Which part(s) of the docs might be affected or should be updated? And how?

https://www.notion.so/risingwave-labs/The-new-architecture-doc-WIP-802b855b37194367a6c905ccc4267449?pvs=4

### Reference

_No response_

### PR DESCRIPTION
**Pull request by hengm3467**
## Info

- **Description**

As the new architecture was only updated in the dev branch, I'm replicating the changes in v1.3.

- **Notes**

  - [ Include any supplementary context or references here. ]

- **Related code PR**

  - [ Provide a link to the relevant code PR here, if applicable. ]

- **Related doc issue**
  
  Resolves #1396

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

File: versioned_docs/version-1.3/Architecture.md [hunk 1]
```diff
@@ -3,20 +3,18 @@ sidebar_position: 3
...
-RisingWave includes these key components:
+The architecture of RisingWave is illustrated in the diagram below. It comprises three main components: a meta node, compute nodes, and compact nodes.
 
-* A serving layer that parses SQL queries and performs planning and optimizations of query jobs.
-* A processing layer that performs all the data computation and status updates.
-* A metadata management service that manages the metadata of different nodes, and coordinates operations among these nodes. It consists of a local metadata store and various management modules.
-* A storage layer that stores data to and retrieves data from object storage like S3.
+The meta node takes charge of managing the metadata of compute and compact nodes and orchestrating operations across the system.
+The compute nodes are responsible for ingesting data from upstream systems, parsing and running SQL queries, and delivering data to downstream systems.
+The compact nodes handle data storage and retrieval from object storage. They also perform data compaction to optimize storage efficiency.
```
Comment (neverchanje) [inline versioned_docs/version-1.3/Architecture.md:15]: Not true. It's just designed for compaction. Data retrieval from object stores doesn't have to go through compact nodes.

File: versioned_docs/version-1.3/Architecture.md [hunk 2]
```diff
@@ -3,20 +3,18 @@ sidebar_position: 3
...
 
-RisingWave includes these key components:
+The architecture of RisingWave is illustrated in the diagram below. It comprises three main components: a meta node, compute nodes, and compact nodes.
 
-* A serving layer that parses SQL queries and performs planning and optimizations of query jobs.
-* A processing layer that performs all the data computation and status updates.
-* A metadata management service that manages the metadata of different nodes, and coordinates operations among these nodes. It consists of a local metadata store and various management modules.
-* A storage layer that stores data to and retrieves data from object storage like S3.
+The meta node takes charge of managing the metadata of compute and compact nodes and orchestrating operations across the system.
+The compute nodes are responsible for ingesting data from upstream systems, parsing and running SQL queries, and delivering data to downstream systems.
```
Comment (neverchanje) [inline versioned_docs/version-1.3/Architecture.md:14]: I am not sure if we still need frontend for now.  Let me check it

Comment (hengm3467) [reply versioned_docs/version-1.3/Architecture.md:14]: @neverchanje What is the evaluation? :)

Comment (fuyufjh) [reply versioned_docs/version-1.3/Architecture.md:14]: > I am not sure if we still need frontend for now. 

I think still yes? 

@hengm3467 May I ask the motivation of this PR? I just checked https://www.notion.so/risingwave-labs/The-new-architecture-doc-WIP-802b855b37194367a6c905ccc4267449 and the frontend node is still there.

We did have discussed merging frontend node into compute node some days, but it's not done yet.

Comment (kwannoel) [reply versioned_docs/version-1.3/Architecture.md:14]: Our webpage seems to already reflect this: https://docs.risingwave.com/docs/current/architecture/

Comment (kwannoel) [reply versioned_docs/version-1.3/Architecture.md:14]: Seems like this was merged in previously: https://github.com/risingwavelabs/risingwave-docs/pull/1412/files

And this PR is just to migrate it over.

**Comment by neverchanje** [comment]
Seems the current page https://docs.risingwave.com/docs/current/architecture/ is satisfying enough. You may want to just close this PR.

**Pull request closed without merging**